Sources/AppcuesKit/Presentation/UI/Components/AppcuesOptionSelect.swift

```diff
@@ -55,19 +61,30 @@ internal struct AppcuesOptionSelect: View {
     @ViewBuilder var items: some View {
         let primaryColor = stepState.shouldShowError(for: model.id) ? Color(dynamicColor: model.errorLabel?.style?.foregroundColor) : nil
 
-        ForEach(model.options) { option in
-            let binding = stepState.formBinding(for: model.id, value: option.value)
-            SelectToggleView(
-                selected: binding,
-                primaryColor: primaryColor,
-                model: model
-            ) {
-                if binding.wrappedValue {
-                    (option.selectedContent ?? option.content).view
-                } else {
-                    option.content.view
-                }
+        // each option and its selection state
+        let selections: [Binding<Bool>] = model.options.map { stepState.formBinding(for: model.id, value: $0.value) }
+
+        // determine if leading fill is supported for items prior to selected item
+        let allowLeadingFill = (model.leadingFill ?? false) && model.selectMode == .single && model.controlPosition == .hidden
+
+        // first item, if any, selected - to check for leading fill
+        let firstSelectedIndex = selections.firstIndex { $0.wrappedValue }
+
+        let optionItems: [OptionItem] = model.options.enumerated().map { index, item in
+            let isSelected = selections[index]
+            var content = (isSelected.wrappedValue) ? (item.selectedContent ?? item.content) : item.content
+            if allowLeadingFill,
+                let firstSelectedIndex = firstSelectedIndex,
+                index < firstSelectedIndex,
+                let selectedContent = item.selectedContent {
+                // this overrides an unselected item with selected content styling in the leadingFill case
+                content = selectedContent
             }
+            return OptionItem(id: item.id, content: content, isSelected: isSelected)
+        }
```

Comment:
I rewrote this block a couple of times trying to find a more elegant way to express this concept inside this `@ViewBuilder` block 😅 - open to any ideas to improve or simplify. It just requires the full selection state and sequence of items to apply that leading fill correctly, rather than being focused on each specific item only as before.